src/tqec/templates/rpng.py

```diff
@@ -0,0 +1,81 @@
+from dataclasses import dataclass
+
+from tqec.enums import Orientation
+from tqec.plaquette.frozendefaultdict import FrozenDefaultDict
+from tqec.plaquette.rpng import RPNGDescription
+from tqec.position import Displacement, Position2D, Shape2D
+from tqec.scale import Scalable2D
+from tqec.templates.indices.base import Template
+
+
+@dataclass
+class RPNGTemplate:
+    template: Template
+    mapping: FrozenDefaultDict[int, RPNGDescription]
+
+    def instantiate(self, k: int) -> list[list[RPNGDescription]]:
+        indices = self.template.instantiate(k)
+        return [[self.mapping[i] for i in row] for row in indices]
+
+    def shape(self, k: int) -> Shape2D:
+        """Returns the current template shape."""
+        return self.template.shape(k)
+
+    @property
+    def scalable_shape(self) -> Scalable2D:
+        """Returns a scalable version of the template shape."""
+        return self.template.scalable_shape
+
+    def get_midline_plaquettes(
+        self, k: int, orientation: Orientation = Orientation.HORIZONTAL
+    ) -> list[tuple[int, int]]:
```

Comment:
In the current implementation of observable construction( `compile/observables`), the `get_midline_plaquettes` method in the `Template` class, as well as the concept of observable qubits, are not used at all. I am wondering whether these components are still needed.

Reply:
I'll open an issue for that. The `Interval` class is also unused at the moment, so that calls for a removal of dead code.

Reply:
See #424